dbt_meshify/main.py

```diff
@@ -48,26 +53,68 @@ def operation():
 
 
 @cli.command(name="connect")
-@click.argument("projects-dir", type=click.Path(exists=True), default=".")
-def connect(projects_dir):
+@project_paths
+@projects_dir
+@exclude_projects
+@read_catalog
+def connect(
+    project_paths: tuple, projects_dir: Path, exclude_projects: List[str], read_catalog: bool
+):
     """
-    !!! info
-        This command is not yet implemented
 
     Connects multiple dbt projects together by adding all necessary dbt Mesh constructs
-    """
-    holder = DbtProjectHolder()
-
-    while True:
-        path_string = input("Enter the relative path to a dbt project (enter 'done' to finish): ")
-        if path_string == "done":
-            break
-
-        path = Path(path_string).expanduser().resolve()
-        project = DbtProject.from_directory(path, read_catalog)
-        holder.register_project(project)
 
-    print(holder.project_map())
+    """
+    if project_paths and projects_dir:
+        raise click.BadOptionUsage(
+            option_name="project_paths",
+            message="Cannot specify both project_paths and projects_dir",
+        )
+    # 1. initialize all the projects supplied to the command
+    # 2. compute the dependency graph between each combination of 2 projects in that set
+    # 3. for each dependency, add the necessary dbt Mesh constructs to each project.
+    #    This includes:
+    #    - adding the dependency to the dependencies.yml file of the downstream project
+    #    - adding contracts and public access to the upstream models
+    #    - deleting the source definition of the upstream models in the downstream project
+    #    - updating the `{{ source }}` macro in the downstream project to a {{ ref }} to the upstream project
+
+    linker = Linker()
+    if project_paths:
+        dbt_projects = [
+            DbtProject.from_directory(project_path, read_catalog) for project_path in project_paths
+        ]
+
+    if projects_dir:
+        dbt_project_paths = [path.parent for path in Path(projects_dir).glob("**/dbt_project.yml")]
+        all_dbt_projects = [
+            DbtProject.from_directory(project_path, read_catalog)
+            for project_path in dbt_project_paths
+        ]
+        dbt_projects = [
+            project for project in all_dbt_projects if project.name not in exclude_projects
+        ]
+
+    dbt_project_combinations = [combo for combo in combinations(dbt_projects, 2)]
+    for dbt_project_combo in dbt_project_combinations:
+        project_map = {project.name: project for project in dbt_project_combo}
+        dependencies = linker.dependencies(dbt_project_combo[0], dbt_project_combo[1])
+        if len(dependencies) == 0:
+            logger.info(
+                f"No dependencies found between {dbt_project_combo[0].name} and {dbt_project_combo[1].name}"
+            )
+        for dependency in linker.dependencies(dbt_project_combo[0], dbt_project_combo[1]):
```

Comment:
No reason to compute this twice.

```suggestion
        for dependency in dependencies:
```